src/components/client/core/Flyout/Flyout.tsx

```diff
@@ -0,0 +1,86 @@
+import { Portal } from "@ark-ui/react";
+import { useState } from "react";
+import { FiX as CloseIcon } from "react-icons/fi";
+
+import Button from "components/client/core/Button/Button";
+import Icon from "components/client/core/Icon/Icon";
+import {
+  Flyout as PrimitiveFlyout,
+  FlyoutArrow,
+  FlyoutArrowTip,
+  FlyoutCloseTrigger,
+  FlyoutContent,
+  FlyoutDescription,
+  FlyoutPositioner,
+  FlyoutTitle,
+  FlyoutTrigger,
+} from "components/primitives";
+import { flyout } from "generated/panda/recipes";
+import { useIsMounted } from "lib/hooks";
+
+import type { FlyoutProps } from "components/primitives";
+import type { ReactNode } from "react";
+
+export interface Props extends FlyoutProps {
+  trigger: ReactNode;
+  title?: string;
+  children: ReactNode;
+}
+
+/**
+ * Core UI flyout.
+ */
+const Flyout = ({ trigger, title, children, ...rest }: Props) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  const classNames = flyout();
+
+  const isMounted = useIsMounted();
+
+  if (!isMounted) return null;
+
+  return (
+    <PrimitiveFlyout
+      open={isOpen}
+      onClose={() => setIsOpen(false)}
+      portalled
+      {...rest}
+    >
+      <FlyoutTrigger asChild onClick={() => setIsOpen(!isOpen)}>
+        {trigger}
+      </FlyoutTrigger>
+
+      <Portal>
+        <FlyoutPositioner className={classNames.positioner}>
+          <FlyoutContent className={classNames.content}>
+            <FlyoutArrow className={classNames.arrow}>
+              <FlyoutArrowTip className={classNames.arrowTip} />
+            </FlyoutArrow>
+            {title && (
+              <FlyoutTitle className={classNames.title}>{title}</FlyoutTitle>
+            )}
+            <FlyoutDescription className={classNames.description}>
+              {children}
+            </FlyoutDescription>
+
+            <FlyoutCloseTrigger asChild onClick={() => setIsOpen(false)}>
+              <Button
+                pos="absolute"
+                top={2}
+                right={2}
+                bgColor={{
+                  base: "inherit",
+                  _hover: "bg.emphasized",
+                }}
```

Comment:
I would also update top to be `1`, to match surrounding padding with container cohesively, and maybe add `p={2}` just to make the button a bit smaller (that is more of a personal preference though).

Reply:
Modifying the code to use the Icon and closeTrigger recipe classes did modify the original button styling so we would need to check other component recipes before making this switch it seems. I added a padding of `p={2}` but I removed the content margin to keep the padding-top and right symmetrical!

Reply:
Yeah we would definitely need to look at it on a case by case basis and adjust what we need to! I just know we have been running into issues with `asChild` prop and I think it is worth doing that overhaul instead of forcing our custom `Button` in there and adding/overriding a bunch of styles. What are your thoughts on that?

Reply:
I totally agree. I like this new approach better and with the added benefit of not using the asChild prop.